app/Http/Controllers/Web/Admin/RockTheVoteImportController.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace App\Http\Controllers\Web\Admin;
+
+use App\Http\Controllers\Controller;
+use App\Models\ImportFile;
+use App\Models\RockTheVoteLog;
+use App\Types\ImportType;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+class RockTheVoteImportController extends Controller
+{
+    /**
+     * The type of import data.
+     *
+     * @var string
+     */
+    public $importType;
+
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('auth:web');
+        $this->middleware('role:admin,staff');
+
+        $this->importType = ImportType::$rockTheVote;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $imports = ImportFile::where('import_type', $this->importType)
+            ->orderBy('id', 'desc')
+            ->paginate(15);
+
+        return view('admin.imports.rock-the-vote.index', [
+            'imports' => $imports,
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $config = ImportType::getConfig($this->importType);
+
+        if (request('source') === 'test') {
+            return $this->renderTestView($config);
+        }
```

Comment:
For later – let's pull this out into a separate `TestRockTheVoteImportController` so we don't have this awkward "two routes in one" flow, since they're really doing very different things.

```suggestion
        // TODO: Extract this into a separate controller!
        if (request('source') === 'test') {
            return $this->renderTestView($config);
        }
```

Reply:
Ya, I like that idea 👍